core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -71,6 +116,15 @@ class MultiCloudWorkspaceService(userInfo: UserInfo,
 
     val workspaceId = UUID.randomUUID
     for {
+      _ <- samDAO.userHasAction(SamResourceTypeNames.billingProject, workspaceRequest.namespace, SamBillingProjectActions.createWorkspace, userInfo).flatMap {
```

Comment:
Adding a check here to ensure the user has the appropriate permissions against the billing project.